articles/css-flexbox-vs-css-grid/index.md

```diff
@@ -0,0 +1,767 @@
+</div>
+```
+
+Some CSS to style the header:
+
+```css
+.nav {
+    display: flex;
+    background: seagreen;
+    height: 50px;
+    align-items: center;
+    gap: 10px;
+}
+
+.nav div {
+    color: whitesmoke;
+    margin-left: 20px;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-header-nav.png)
+
+Most common header layouts lay their items in one direction; thus, it's much easier to use Flexbox because it arranges items in one direction.
+
+On the other hand, a Grid is a two-dimensional layout, meaning you can lay out items in rows and columns.
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid.png)
+[***Image Source***](https://www.youtube.com/watch?v=MdPYenT89o4)
+
+An example of a basic two-dimensional layout.
+
+```html
+<div class="container">
+    <div class="item item1"></div>
+    <div class="item item2"></div>
+    <div class="item item3"></div>
+    <div class="item item4"></div>
+    <div class="item item5"></div>
+    <div class="item item6"></div>
+</div>
+```
+
+```css
+.container {
+    display: grid;
+    grid-template-columns: 150px 150px 150px;
+    grid-gap: 1rem;
+    grid-template-rows: 150px 150px;
+}
+
+.item {
+    background-color: #1EAAFC;
+    color: #fff;
+    border: 6px solid #171717;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid-example.jpg)
+
+We specify a grid with `display: grid` and then determine rows and columns with `grid-template-columns` and `grid-template-rows`. This defines a two-dimensional layout. In this case, each column and row take the size of `150px`.
+
+#### Content vs layout
+
+Flexbox is content-based so content comes first. By content first, we mean it helps you align the contents of and move blocks and items within a container.
+
+On the other hand, the CSS Grid focuses on laying out the outer/skeleton layout of the page.
+
+To understand the content vs layout concept, let's go back to the header navigation bar design we've discussed above. We can target an item inside the header and determine where to display it.
+
+To get the [core difference](https://www.youtube.com/watch?v=9zA8cB-54SA), Assuming we want to move the `Sign In` and `Sign Up` menu to the left, we can do it using Flexbox and CSS Grid.
+
+Using Flexbox;
+
+```css
+.nav {
+    display: flex;
+    background: seagreen;
+    height: 50px;
+    align-items: center;
+    gap: 1px;
+}
+
+.nav div {
+    margin-left: 20px;
+    margin-right: 20px;
+    color: whitesmoke;
+}
+
+di v:nth-child(4) {
+    margin-left: auto;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-content-first.png)
+
+We specify the child item of the container and determine its position. That's what we mean by referring to Flexbox as content-based.
+
+On the other hand, we can implement the same using CSS Grid. The header is one dimensional, and Flexbox is perfect for doing this, but we can do that with CSS Grid instead to differentiate between layout and content.
+
+```css
+.nav-grid {
```

Comment:
@kimkimani, before you used the .nav class for the HTML but now you are using .nav-grid. You should mention this change to readers.